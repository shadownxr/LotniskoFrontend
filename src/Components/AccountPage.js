import React from 'react';

export default function AccountPage(props){
    return(
        <div>
            <div className="Account">
                Username: {props.accountData.username} <br/>
                Email: {props.accountData.email} <br/>
            </div>
        </div>
    )
}