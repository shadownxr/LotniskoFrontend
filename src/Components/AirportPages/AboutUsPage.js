import React from 'react';

export default function AboutUsPage(){
    return(
        <div>
            <div className="AboutUs">AboutUs</div>
        </div>
    )
}