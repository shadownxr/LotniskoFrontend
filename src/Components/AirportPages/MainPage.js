import React from 'react';

export default function MainPage(){
    return(
        <div>
            <div className="MainPage">MainPage</div>
        </div>
    )
}