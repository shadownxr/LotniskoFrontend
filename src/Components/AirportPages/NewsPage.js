import React from 'react';

export default function NewsPage(){
    return(
        <div>
            <div className="News">NewsPage</div>
        </div>
    )
}