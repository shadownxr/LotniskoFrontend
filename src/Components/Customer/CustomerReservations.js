import React, { useState,useEffect, useRef } from "react";
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import SearchButton from './SearchReservationButton';
import CustomerReservationsList from './CustomerReservationsList';
import Cookie from 'react-cookies';

const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

const useStyles = makeStyles({
    table: {
      minWidth: 650,
    },
});

/**
 * Renders customers reservations
 * @param {accountData} props 
 */
export default function CustomerReservations(props){
    const [reservations,setReservations] = useState([]);
    const [search, setSearch] = useState("");
    const [refresh,setRefresh] = useState(false);
    const [searchedReservations,setSearchedReservations] = useState([]);

    const ref = useRef();
    ref.current = "";

    useEffect(() => {
      fetchReservations();
    },[]);

    useEffect(() => {
      if(refresh === true){
        fetchReservations();
        setRefresh(false);
      }
    },[refresh])

    useEffect(() => {
      if(ref !== search){
        handleSearch();
      }
    },[search,ref])

    const handleSearch = () => {
      let searched = reservations.filter((reservation) => {
        return(
          (reservation.flightID.sapid.cityName === search.from) && 
          (reservation.flightID.dapid.cityName === search.to) && 
          (new Date(reservation.startDate).toLocaleDateString() >= new Date(search.dateFrom).toLocaleDateString()) &&
          //(new Date(reservation.startDate).toLocaleDateString() <= new Date(search.dateTo).toLocaleDateString()) &&
          (reservation.paid === search.paid)
        )
      }).map((reservation) => reservation);
      setSearchedReservations(searched);
    }

    const fetchReservations = () => {
        let payload = {
          "userID": props.accountData.id,
        }

      const url = "https://localhost:8443/api/tickets/listByClientID";

      const options = {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Authorization': 'Bearer ' + Cookie.load('userToken').token,
          },
          body: JSON.stringify(payload),
      };

      fetch(url, options)
        .then(response => response.json())
        .then(result => {
          setReservations(result);
        });
    }

    return(
        <div>
            <TableContainer component={Paper}>
                <Table className={useStyles().table} aria-label="simple table">
                    <TableHead>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={8}>Moje rezerwacje</StyledTableCell>
                        </StyledTableRow>
                        <StyledTableRow>
                            <StyledTableCell align="center">Z</StyledTableCell>
                            <StyledTableCell align="center">Do</StyledTableCell>
                            <StyledTableCell align="center">Data wylotu</StyledTableCell>
                            <StyledTableCell align="center">Klasa</StyledTableCell>
                            <StyledTableCell align="center">Cena</StyledTableCell>
                            <StyledTableCell align="center">Przeprowadzona odprawa</StyledTableCell>
                            <StyledTableCell align="center"><SearchButton search={(search) => {setSearch(search)}}/></StyledTableCell>
                        </StyledTableRow>
                    </TableHead>
                    <CustomerReservationsList customerReservationsData={(search === "")?reservations:searchedReservations} refresh={(refresh) => {setRefresh(true)}}/>
                </Table>
            </TableContainer>
        </div>
    )
}