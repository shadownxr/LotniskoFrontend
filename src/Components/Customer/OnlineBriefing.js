import React, { useState } from 'react';
import BriefableReservations from './BriefableReservations';
import UserDataForm from './UserDataForm';
import TicketPrinter from './TicketPrinter';
import SeatChooser from './SeatChooser';

export default function OnlineBriefing(props){
    const [briefingScreen, setBriefingScreen] = useState(1);
    const [reservation, setReservation] = useState([]);
    const [ticketData, setTicketData] = useState({});
    const [seat, setSeat] = useState("");
    const [refresh, setRefresh] = useState(false);

    const briefing = () => {
        switch(briefingScreen){
            case 1:
                return <BriefableReservations accountData={props.accountData} refresh={refresh} reservation={(reservation) => {setReservation(reservation)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            case 2:
                return <UserDataForm screen={(screen) => {setBriefingScreen(screen)}} ticketData={(ticketData) => {setTicketData(ticketData)}}/>;
            case 3:
                return <SeatChooser reservation={reservation} seat={(seat) => {setSeat(seat)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            case 4:
                return <TicketPrinter reservation={reservation} seat={seat} ticketData={ticketData} refresh={(refresh) => {setRefresh(refresh)}} screen={(screen) => {setBriefingScreen(screen)}}/>;
            default:
                return <BriefableReservations accountData={props.accountData} reservation={(reservation) => {setReservation(reservation)}} screen={(screen) => {setBriefingScreen(screen)}} refresh={(refresh) => {setRefresh(refresh)}}/>;
        }
    }

    return(
        <div>{briefing()}</div>
    )
}