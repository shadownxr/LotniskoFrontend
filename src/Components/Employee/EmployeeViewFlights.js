import React from 'react';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import SearchButton from '../Flights/SearchFlightButton';
import EmployeeViewFlightsList from './EmployeeViewFlightsList';
import AddButton from './AddFlightButton';

const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

const useStyles = makeStyles({
    table: {
      minWidth: 650,
    },
});

export default function EmployeeViewFlights(){
    const flights = [{
        id: 1,
        from: "Szczecin",
        to: "Krak贸w",
        date: "2020-07-02",
        class: "Ekonomiczna",
        cost: "200PLN" 
     },{
        id: 2,
        from: "Krak贸w",
        to: "Rak贸w",
        date: "2020-07-20",
        class: "Biznes",
        cost: "400PLN"
     }]

    return(
        <div>
            <TableContainer component={Paper}>
                <Table className={useStyles().table} aria-label="simple table">
                    <TableHead>
                        <StyledTableRow>
                            <StyledTableCell align="center" colSpan={7}>Lista lot贸w</StyledTableCell>
                        </StyledTableRow>
                        <StyledTableRow>
                            <StyledTableCell align="center">Nr</StyledTableCell>
                            <StyledTableCell align="center">Z</StyledTableCell>
                            <StyledTableCell align="center">Do</StyledTableCell>
                            <StyledTableCell align="center">Data wylotu</StyledTableCell>
                            <StyledTableCell align="center">Klasa</StyledTableCell>
                            <StyledTableCell align="center">Cena</StyledTableCell>
                            <StyledTableCell align="right"><div style={{display:"inline-flex"}}><SearchButton /><AddButton /></div></StyledTableCell>
                        </StyledTableRow>
                    </TableHead>
                    <EmployeeViewFlightsList flightsData={flights}/>
                </Table>
            </TableContainer>
        </div>
    )
}