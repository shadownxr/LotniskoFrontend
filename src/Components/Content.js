import React from 'react';
import NewsPage from './AirportPages/NewsPage';
import AboutUsPage from './AirportPages/AboutUsPage';
import MainPage from './AirportPages/MainPage';
import AccountPage from './AccountPage';
import ViewFlights from './Flights/ViewFlights';
import OnlineBriefing from './Customer/OnlineBriefing';
import EmployeeViewFlights from './Employee/ManageFlights/EmployeeViewFlights';
import ViewEmployees from './Manager/AdministerEmployees/ViewEmployees';
import ViewFiredEmployees from "./Manager/AdministerEmployees/ViewFiredEmployees";
import CustomerReservations from './Customer/CustomerReservations';
import EmployeeViewPlanes from "./Employee/ManagePlanes/EmployeeViewPlanes";
import EmployeeViewAirports from "./Employee/ManageAirports/EmployeeViewAirports";

/**
 * Component that render pages in Content div depending on user choice
 * @param {menuChoice,accountData} props 
 */
export default function Content(props){
    const renderContentSwitch = () => {
        switch(props.menuChoice){
          case "Main Page":
            return <MainPage />;
          case "News":
            return <NewsPage />;
          case "About Us":
            return <AboutUsPage />;
          case "Account Page":
            return <AccountPage accountData={props.accountData}/>;
          case "View Flights":
            return <ViewFlights accountData={props.accountData}/>;
          case "Online Briefing":
            return <OnlineBriefing accountData={props.accountData}/>;
          case "Employee View Flights":
            return <EmployeeViewFlights accountData={props.accountData}/>;
          case "Employee View Airports":
            return <EmployeeViewAirports accountData={props.accountData}/>;
          case "Employee View Planes":
            return <EmployeeViewPlanes accountData={props.accountData}/>;
          case "View Employees":
            return <ViewEmployees accountData={props.accountData}/>;
          case "View Fired Employees":
            return <ViewFiredEmployees accountData={props.accountData}/>;
          case "Customer Reservations":
            return <CustomerReservations accountData={props.accountData}/>;
          default:
            return <MainPage />;
        }
      }

      return(
        <div>
          {renderContentSwitch()}
        </div>
      )
}
