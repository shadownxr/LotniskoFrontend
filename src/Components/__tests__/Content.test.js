import React from 'react';
import { shallow } from 'enzyme';

import MainPage from "../AirportPages/MainPage";
import Content from "../Content";
import NewsPage from "../AirportPages/NewsPage";
import AboutUsPage from "../AirportPages/AboutUsPage";
import AccountPage from "../AccountPage";
import ViewFlights from "../Flights/ViewFlights";
import OnlineBriefing from "../Customer/OnlineBriefing";
import EmployeeViewFlights from "../Employee/ManageFlights/EmployeeViewFlights";
import EmployeeViewAirports from "../Employee/ManageAirports/EmployeeViewAirports";
import EmployeeViewPlanes from "../Employee/ManagePlanes/EmployeeViewPlanes";
import ViewEmployees from "../Manager/AdministerEmployees/ViewEmployees";
import ViewFiredEmployees from "../Manager/AdministerEmployees/ViewFiredEmployees";
import CustomerReservations from "../Customer/CustomerReservations";


const accountData = {username: "Admin", email: "Admin@Example.com"};

describe('UserMenu test',() =>{
    it('Renders when loaded',() =>{
        const wrapper = shallow(<Content />);
        expect(wrapper.find(<MainPage />)).toMatchSnapshot();
    })
    it('Renders Main Page menu on change',() => {
        const choice = "Main Page";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<MainPage />)).toMatchSnapshot();
    })
    it('Renders Main Page menu on change',() => {
        const choice = "Main Page";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<MainPage />)).toMatchSnapshot();
    })
    it('Renders NewsPage menu on change',() => {
        const choice = "News";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<NewsPage />)).toMatchSnapshot();
    })
    it('Renders AboutUsPage menu on change',() => {
        const choice = "About Us";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<AboutUsPage />)).toMatchSnapshot();
    })
    it('Renders AccountPage on change',() => {
        const choice = "Account Page";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<AccountPage accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders ViewFlights on change',() => {
        const choice = "View Flights";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<ViewFlights accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders OnlineBriefing on change',() => {
        const choice = "Online Briefing";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<OnlineBriefing accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders OnlineBriefing menu on change',() => {
        const choice = "Online Briefing";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<OnlineBriefing accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders EmployeeViewFlights menu on change',() => {
        const choice = "Employee View Flights";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<EmployeeViewFlights accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders EmployeeViewAirportsmenu on change',() => {
        const choice = "Employee View Airports";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<EmployeeViewAirports accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders EmployeeViewPlanes menu on change',() => {
        const choice = "Employee View Planes";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<EmployeeViewPlanes accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders ViewEmployees menu on change',() => {
        const choice = "View Employees";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<ViewEmployees accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders ViewFiredEmployees menu on change',() => {
        const choice = "View Fired Employees";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<ViewFiredEmployees accountData={accountData}/>)).toMatchSnapshot();
    })
    it('Renders CustomerReservations menu on change',() => {
        const choice = "Customer Reservations";
        const wrapper = shallow(<Content menuChoice={choice}/>);
        expect(wrapper.find(<CustomerReservations accountData={accountData}/>)).toMatchSnapshot();
    })
})
