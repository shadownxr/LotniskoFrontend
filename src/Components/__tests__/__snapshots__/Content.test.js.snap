// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu test Renders AboutUsPage menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders AccountPage on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders CustomerReservations menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders EmployeeViewAirportsmenu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders EmployeeViewFlights menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders EmployeeViewPlanes menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders Main Page menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders Main Page menu on change 2`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders NewsPage menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders OnlineBriefing menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders OnlineBriefing on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders ViewEmployees menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders ViewFiredEmployees menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders ViewFlights on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders when loaded 1`] = `ShallowWrapper {}`;
